pipelines/gene_symbol_classifier/GeneSymbolClassifier_production_conf.pm

```diff
@@ -42,59 +42,41 @@ use Bio::EnsEMBL::ApiVersion qw/software_version/;
 use Bio::EnsEMBL::Hive::Version 2.5;
 use Bio::EnsEMBL::Hive::PipeConfig::HiveGeneric_conf;
 
+use File::Spec::Functions qw(catdir catfile);
+
 sub default_options {
   my ($self) = @_;
 
-  # inherited from the Base pipeline config
-  my $gsc_data_directory = $self->o('pipeline_dir');
-
-  #my $pipeline_name => $self->o('pipeline_name') || 'GeneSymbolClassifier_production';
-  my $pipeline_name => 'GeneSymbolClassifier_production';
-  my $pipe_db_name = $ENV{USER}.$pipeline_name;
-
-  #my $gsc_scripts_directory = $ENV{ENSCODE}.'/ensembl-genes/pipelines/gene_symbol_classifier';
-  my $gsc_scripts_directory = '/nfs/production/flicek/ensembl/genebuild/william/gene-symbol-classifier/production-pipeline/ensembl-genes/pipelines/gene_symbol_classifier';
-
-  #my $protein_sequences_fasta_path = "${gsc_data_directory}/${core_db_name}_protein_sequences.fa";
-  #my $gene_symbols_csv_path = "${gsc_data_directory}/${core_db_name}_protein_sequences_symbols.csv";
-  #my $filtered_assignments_csv_path = "${gsc_data_directory}/${core_db_name}_protein_sequences_symbols_filtered.csv";
-
   return {
     %{$self->SUPER::default_options},
 
     species      => [],
-    #taxon        => [],
+    taxon        => [],
     division     => [],
-    run_all      => 0,
+    run_all      => 1,
     antispecies  => [],
-    #antitaxons   => [],
+    antitaxons   => [],
     meta_filters => {},
 
     history_file => undef,
 
-    pipeline_name => $pipeline_name,
-
-    pipeline_db => {
-      -driver => 'mysql',
-      -host   => $self->o('host'),
-      -port   => $self->o('port'),
-      -user   => $self->o('user'),
-      -pass   => $self->o('password'),
-      -dbname => $self->o('pipe_db_name'),
-    },
-
-    user_r => 'ensro',
-
-    loading_threshold => $self->o('loading_threshold'),
-
-    gsc_scripts_directory => $gsc_scripts_directory,
-    gsc_data_directory => $gsc_data_directory,
-    #protein_sequences_fasta_path => $protein_sequences_fasta_path,
-    #gene_symbols_csv_path => $gene_symbols_csv_path,
-    #filtered_assignments_csv_path => $filtered_assignments_csv_path,
-
-    #core_db_server_host => $core_db_server_host,
-    #core_db_server_port => $core_db_server_port,
+    pipeline_name => 'GeneSymbolClassifier_production',
+
+    loading_threshold => 0.7,
+
+    old_server_uri => '',
+    ensembl_root_dir => $ENV{ENSEMBL_ROOT_DIR},
+    scripts_directory => catdir($self->o('ensembl_root_dir'), 'ensembl-genes', 'pipelines', 'gene_symbol_classifier'),
+    gsc_image_version => '0.12.1',
+    filter_gsc_image_version => '0.3.0',
+    classifier_filename => 'mlp_10_min_frequency_2022-01-29_03.08.32.ckpt',
```

Comment:
Potentially avoid hardcoding values here?

Reply:
I see this as defaults. If you want to use a different option people can provide as parameter `--classifier_filename <whatever>` when initialising the pipeline
For me hardcoding would be something like `cmd => 'singularity run docker://ensemblorg/gsc_pipeline:0.3.0'`

Reply:
I see, personally I would try to avoid adding any values that point to specific data files (although still happens sometimes), but I think it's nitpicking in this case, no strong reason to change it.